src/quantum_object.jl

```diff
@@ -717,6 +717,110 @@ Quantum Object:   type=Operator   dims=[20, 20]   size=(400, 400)   ishermitian=
 """
 ⊗(A::QuantumObject, B::QuantumObject) = kron(A, B)
 
+@doc raw"""
+    permute(A::QuantumObject, order::Vector{Int})
+
+Permute the subsystems of a [`QuantumObject`](@ref) `A` according to the order specified in `order`. So, if `order = [2, 1, 3]`, the first subsystem will become the second, the second will become the first, and the third will remain the same, i.e. H₁ ⊗ H₂ ⊗ H₃ → H₂ ⊗ H₁ ⊗ H₃.
+
+This method currently dispatches only for [`Ket`](@ref), [`Bra`](@ref), and [`Operator`](@ref) types of [`QuantumObject`](@ref).
+
+# Examples
+
+```julia
+ψ1 = fock(2, 0)
+ψ2 = fock(2, 1)
+ψ_12 = tensor(ψ1, ψ2)
+
+permute(ψ_12, [2, 1]) ≈ tensor(ψ2, ψ1)
+
+# Returns true
+```
+
+"""
+function permute end
+
+#  -------------------------------------------------------------
+#  helper functions
+#  -------------------------------------------------------------
+
+function kron_rep(
+    dims;
+    labels="abcdefghijklmnopqrstuvwxyz"[1:length(dims)]
+)
+
+    # create string representations of the subsystems
+    # and store them in the order of the permutation
+    xs = []
+    for i = 1:length(dims)
+        x_i = labels[i] .* string.(1:dims[i])
+        if i != length(dims)
+            x_i .*= "_"
+        end
+        push!(xs, x_i)
+    end
+
+    return kron(xs...)
+end
+
+function kron_permutation(perm::Vector{Int}, dims::Vector{Int})
```

Comment:
I suggest this to be an internal function, can you rename this as `_kron_permutation`